kratos/elements/levelset_convection_element_simplex_algebraic_stabilization.cpp

```diff
@@ -0,0 +1,220 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:        BSD License
+//                  Kratos default license: kratos/license.txt
+//
+//  Main authors:   Mohammad R. Hashemi
+//
+//
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "elements/levelset_convection_element_simplex_algebraic_stabilization.h"
+
+
+namespace Kratos
+{
+    template< unsigned int TDim, unsigned int TNumNodes>
+    LevelSetConvectionElementSimplexAlgebraicStabilization<TDim, TNumNodes>::LevelSetConvectionElementSimplexAlgebraicStabilization() 
+    : LevelSetConvectionElementSimplex<TDim, TNumNodes>::LevelSetConvectionElementSimplex()
+    {}
+
+    template< unsigned int TDim, unsigned int TNumNodes>
+    LevelSetConvectionElementSimplexAlgebraicStabilization<TDim, TNumNodes>::LevelSetConvectionElementSimplexAlgebraicStabilization(
+        IndexType NewId,
+        GeometryType::Pointer pGeometry)
+    : LevelSetConvectionElementSimplex<TDim, TNumNodes>::LevelSetConvectionElementSimplex(NewId, pGeometry)
+    {}
+
+    template< unsigned int TDim, unsigned int TNumNodes>
+    LevelSetConvectionElementSimplexAlgebraicStabilization<TDim, TNumNodes>::LevelSetConvectionElementSimplexAlgebraicStabilization(
+        IndexType NewId,
+        GeometryType::Pointer pGeometry,
+        PropertiesType::Pointer pProperties)
+    : LevelSetConvectionElementSimplex<TDim, TNumNodes>::LevelSetConvectionElementSimplex(NewId, pGeometry, pProperties)
+    {}
+
+    /// Destructor.
+    template< unsigned int TDim, unsigned int TNumNodes>
+    LevelSetConvectionElementSimplexAlgebraicStabilization<TDim, TNumNodes>::~LevelSetConvectionElementSimplexAlgebraicStabilization() {}
+
+    template< unsigned int TDim, unsigned int TNumNodes>
+    Element::Pointer LevelSetConvectionElementSimplexAlgebraicStabilization<TDim, TNumNodes>::Create(
+        IndexType NewId,
+        NodesArrayType const& ThisNodes,
+        PropertiesType::Pointer pProperties) const
+    {
+        KRATOS_TRY
+        return Element::Pointer(new LevelSetConvectionElementSimplexAlgebraicStabilization(NewId, this->GetGeometry().Create(ThisNodes), pProperties));
+        KRATOS_CATCH("");
+    }
+
+    template< unsigned int TDim, unsigned int TNumNodes>
+    void LevelSetConvectionElementSimplexAlgebraicStabilization<TDim, TNumNodes>::CalculateLocalSystem(
+        MatrixType &rLeftHandSideMatrix,
+        VectorType &rRightHandSideVector,
+        const ProcessInfo &rCurrentProcessInfo)
+    {
```

Comment:
this is a pretty large fct, consider splitting it up

Reply:
Having a look to the formulation, I think that right now it is way easy to follow. Considering that the function is around 150 lines (and that good practices say around 100) I'd keep it as it is for the sake of readability.

Maybe adding some inline comments can clarify what is done there.